### PR TITLE
Add tuning setup

### PR DESCRIPTION
**Pull request by JanAnders**
This PR adds the code for tuning the model in the training container.
It can be specified for a given runtime or number of tried hyperparams and also creates a pareto fron visualization.
The ability to configure the tuning and export of the pareto-front plot still have to be added.

### CODE REVIEW

File: src/models/tuning/optim.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+from src.models.tuning.configspace import configspace
+from src.models.train_model import train, load_data, evaluate_model
+from src.models.tuning.plot_pareto import plot_pareto
+
+# Global logger
+log_fmt = "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+logging.basicConfig(level=logging.INFO, format=log_fmt)
+logger = logging.getLogger(__name__)
+
+data = load_data("./data/processed/data.csv")
```
Comment (mkoehnenlmu) [inline src/models/tuning/optim.py:25]: i think it would make sense to add this to the config file

File: src/models/tuning/optim.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+
+def evaluate_config(config: Configuration, seed: int = 42) -> float:
+    params = config.get_dictionary()
+
+    print("Evaluate config: " + str(params))
+
+    train_data = data.sample(frac=0.8, random_state=42)
```
Comment (mkoehnenlmu) [inline src/models/tuning/optim.py:33]: random states could also be something worth writing to the config

File: src/models/tuning/optim.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+    print("Loss: " + str(loss.item()))
+
+    return {
+        "rmse": loss.item(),
+        "size": np.sqrt(sum(p.numel() for p in model.parameters())),
+    }
+
+
+def optimize_configuration(cfg):
```
Comment (mkoehnenlmu) [inline src/models/tuning/optim.py:48]: the optimizer configuration should also go into a .ymal - maybe we can apply our hydra knowledge here

**Review by mkoehnenlmu — APPROVED** [review]
Thanks a lot for the work, this should help our model a lot! There is some stuff that could be moved to config files, but this can be also taken care of in a separate branch.

File: src/models/tuning/optim.py (hunk 3 above)
Comment (JanAnders) [reply src/models/tuning/optim.py:48]: Thanks! Started on this, can be improved further.

File: src/models/tuning/optim.py (hunk 2 above)
Comment (JanAnders) [reply src/models/tuning/optim.py:31]: yup, I tried that, didn't get it to work and wasn't high prior enough. We can add this later.

**Comment by mkoehnenlmu** [comment]
test fails on new input size, this will be fixed as soon as model runs in cloud, therefor will be ignored here

**Pull request merged**